### ISSUE TITLE
Add LoadAndUnprotectWallet on wallet plugin

### ISSUE THREAD

**Issue by 0xMazout** [massalabs/thyra-plugin-wallet#35]

**Comment by 0xMazout** [comment]
ADD LoadAndUnprotectWallet on the wallet plugin

When it's done go to thyra and clean the following code

func LoadAndUnprotectWallet(nickname string, app *fyne.App) (*wallet.Wallet, middleware.Responder) {
	wallet, err := wallet.Load(nickname)
	if err != nil {
		return nil, createInternalServerError(errorCodeGetWallet, err.Error())
	}

	clearPassword, err := gui.AskPassword(wallet.Nickname, app)
	if err != nil {
		return nil, createInternalServerError(ErrorCodeWalletCanceledAction, err.Error())
	}

	if len(clearPassword) == 0 {
		return nil, createInternalServerError(ErrorCodeWalletPasswordEmptyWebCreator, ErrorCodeWalletPasswordEmptyWebCreator)
	}

	err = wallet.Unprotect(clearPassword, 0)
	if err != nil {
		return nil, createInternalServerError(errorCodeWalletWrongPassword, err.Error())
	}

	return wallet, nil
}

**Comment by 0xMazout** [comment]
Update implementations in Thyra when you encounter this :

`wallet, _ := LoadAndUnprotectWallet(params.Nickname, app)`

**Comment by SlnPons** [comment]
Hey @0xMazout could you try to use the task template please? 
Here I'm not sure to understand what it does vs. what exist (from a wallet plugin and a thyra wallet perspective + there is this issue https://github.com/massalabs/thyra/issues/382 on thyra side so why should we delete only this part? - hard for me to understand by reading your ticket)
---
It helps to highlight the value of each issue so we can drive delivery via value. 
Also each ticket should be assigned to a PM. And only the PM add them to the board.

To summarise, here is the process that needs to be known and applied: 
1. Anyone can create issues 
2. Created issues needs to follow up the template (if the template is not good anymore, open a PR with suggested changes!) 
3. Assign issue to a PM 
4. Share created issue with the PM

Hope it's clearer ;)

**Comment by 0xMazout** [comment]
Yes understood.

This issue was a reminder to externalize this part in Plugin-wallet. 
as i said in this : https://github.com/massalabs/thyra/pull/442. 
it was a commentary proposal from a reviewer.

And to clarify the situation it was not to delete Thyra but to implement this in plugin wallet . But after verification it seems the logical behind this has changed . So we can close 

(Maded it friday before leaving, so lil bit messy)
